### ISSUE TITLE
(Potential) Reversible RescaleIntensity Transform

### ISSUE THREAD

**Issue by nicoloesch** [fepegar/torchio#993]
**🚀 Feature**
Implement the option to revert the intensity of the RescaleIntensity Transform

**Motivation**

Would be nice to be able to revert the applied transforms to be able to display the images in the original space.

**Pitch**

In the class `RescaleIntensity` in the file `torchio.transforms.preprocessing.intensity.rescale.py`, change the function `rescale` in L.94-96 to overwrite the class attribute `self.in_min_max`:

```python
if self.in_min_max is None:
    self.in_min_max = (array.min(), array.max())
in_min, in_max = self.in_min_max
```

By including the input minimum and maximum values (if not specifically specified in the instantiation of the transform), one can easily revert the min-max transform by replacing L105-109 with the following:

```python
out_range = self.out_max - self.out_min
        if self.invert_transform:
            array -= self.out_min
            array /= out_range
            array *= in_range
            array += in_min
        else:
            array -= in_min
            array /= in_range
            array *= out_range
            array += self.out_min
```

and by providing the class attribute `invert_transform = False` in `__init__`.

**Comments**

This inverse only works as intended if one does not use the percentile approach. In order to revert the clipping, a different approach to restore the initial values would need to be pursued. 

Furthermore, my testing revealed that the inverse transform produces some loss. I utilised a MRI slice of the BraTS challenge of size `[1,1,240,240]` and the inverse produced a total difference of 0.0123 summed across all pixels (2.14e-7 per pixel on average/ double-checked with `torch.mean`). Casting to float of the input array before the forward transformation did not resolve this (most likely requires further investigation) but I assume it is due to floating point errors.

**Comment by fepegar** [comment]
Thanks, @nicoloesch. I think this makes sense. I wouldn't worry about the numerical precision issue. I suspect it would be improved using float64. Would you like to submit a PR?

**Comment by fepegar** [comment]
@nicoloesch there's always a first time! Feel free to open your PR after reading the [contributing guide](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst).

> What is your opinion regarding my comment about the clipping as a result of the percentile method? Is this a consideration for the future as the inverse of the transform would still work but the clipped values remain clipped (but rescaled to the initial space)?

Applying and reverting a transform is sometimes lossy, [as explained in the docs](https://torchio.readthedocs.io/transforms/transforms.html#invertibility). I think it's fine. We could introduce a system where a warning is shown for lossy inversions, but that's probably out of the scope of this PR.

**Comment by fepegar** [comment]
Added in `v0.18.89`.
